tests/test_group.rs

```diff
@@ -150,109 +203,162 @@ fn basic_group_setup() {
-
-    let mut group_bob = Group::new_from_welcome(
-        bob_identity,
-        welcome_bundle_alice_bob.unwrap(),
-        bob_key_package_bundle,
-    );
-
-    assert_eq!(group_alice.tree.nodes, group_bob.tree.nodes);
-    group_alice.tree.print(&format!("\n{:?}", ms1));
-
-    // Alice sends a message to Bob
-    let message_alice = [1, 2, 3];
-    let mls_plaintext_alice = group_alice.create_application_message(&message_alice, None);
-    let encrypted_message = group_alice.encrypt(&mls_plaintext_alice);
-    let mls_plaintext_bob = group_bob.decrypt(&encrypted_message);
-    assert_eq!(mls_plaintext_alice, mls_plaintext_bob);
-
+    use maelstrom::extensions::*;
+    use maelstrom::utils::*;
+    //let ciphersuite_name = CiphersuiteName::MLS10_128_DHKEMX25519_AES128GCM_SHA256_Ed25519;
+    let supported_ciphersuites = vec![
+        CiphersuiteName::MLS10_128_DHKEMX25519_AES128GCM_SHA256_Ed25519,
+        CiphersuiteName::MLS10_128_DHKEMX25519_CHACHA20POLY1305_SHA256_Ed25519,
+    ];
+
+    for ciphersuite_name in supported_ciphersuites {
+        let group_aad = b"Alice's test group";
+
+        // Define identities
+        let alice_credential_bundle =
+            CredentialBundle::new("Alice".into(), CredentialType::Basic, ciphersuite_name).unwrap();
+        let bob_credential_bundle =
+            CredentialBundle::new("Bob".into(), CredentialType::Basic, ciphersuite_name).unwrap();
+
+        // Mandatory extensions
+        let capabilities_extension = Box::new(CapabilitiesExtension::default());
+        let lifetime_extension = Box::new(LifetimeExtension::new(60));
+        let mandatory_extensions: Vec<Box<dyn Extension>> =
+            vec![capabilities_extension, lifetime_extension];
+
+        // Generate KeyPackages
+        let alice_key_package_bundle = KeyPackageBundle::new(
+            ciphersuite_name,
+            &alice_credential_bundle,
+            mandatory_extensions.clone(),
+        );
+
+        let bob_key_package_bundle = KeyPackageBundle::new(
+            ciphersuite_name,
+            &bob_credential_bundle,
+            mandatory_extensions,
+        );
+        let bob_key_package = bob_key_package_bundle.get_key_package();
+
+        // Alice creates a group
+        let group_id = [1, 2, 3, 4];
+        let mut group_alice_1234 =
+            MlsGroup::new(&group_id, ciphersuite_name, alice_key_package_bundle);
+
+        // Alice adds Bob
+        let bob_add_proposal = group_alice_1234.create_add_proposal(
+            group_aad,
+            &alice_credential_bundle,
+            bob_key_package.clone(),
+        );
+        let epoch_proposals = vec![bob_add_proposal];
+        let (mls_plaintext_commit, welcome_bundle_alice_bob_option) = match group_alice_1234
+            .create_commit(
+                group_aad,
+                &alice_credential_bundle,
+                epoch_proposals.clone(),
+                false, // TODO force update seems broken
```

Comment:
Shouldn't the `create_commit_optional_path` test fail then?

Reply:
That comment was outdated